### PR TITLE
Be refactor optimize rowset

### PR DESCRIPTION
**Pull request by kangpinghuang**
Add get_data_disk_size/get_index_disk_size/empty/zero_num_rows/get_num_rows interfaces

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 1]
```diff
@@ -52,11 +52,7 @@ OLAPStatus AlphaRowset::init() {
 }
 
 std::unique_ptr<RowsetReader> AlphaRowset::create_reader() {
-    std::vector<SegmentGroup*> segment_groups;
-    for (auto& segment_group : _segment_groups) {
-        segment_groups.push_back(segment_group.get());
-    }
-    return std::unique_ptr<RowsetReader>(new AlphaRowsetReader(_tablet_schema,
+    return std::unique_ptr<RowsetReader>(new AlphaRowsetReader(
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: unique_ptr cannot be assigned

File: be/src/olap/rowset/alpha_rowset.h [hunk 2]
```diff
@@ -44,14 +44,25 @@ class AlphaRowset : public Rowset {
...
+
+    virtual bool zero_num_rows() const;
+
+    virtual size_t get_num_rows() const;
+
     bool create_hard_links(std::vector<std::string>* success_links);
 
     bool remove_old_files(std::vector<std::string>* removed_links);
 
+
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.h]: empty line

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 3]
```diff
@@ -72,7 +68,7 @@ OLAPStatus AlphaRowset::remove() {
     return OLAP_SUCCESS;
 }
 
-RowsetMetaSharedPtr AlphaRowset::get_meta() {
+RowsetMetaSharedPtr AlphaRowset::get_rs_meta() const {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: May it is better to named as rowset_meta()?

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 4]
```diff
@@ -105,14 +101,34 @@ bool AlphaRowset::remove_old_files(std::vector<std::string>* removed_links) {
...
+bool AlphaRowset::empty() const {
+    return _rowset_meta->empty();
+}
+
+bool AlphaRowset::zero_num_rows() const {
+    return _rowset_meta->row_number() == 0;
+}
+
+size_t AlphaRowset::get_num_rows() const {
+    return _rowset_meta->row_number();
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: The same question like above.

File: be/src/olap/rowset/alpha_rowset_meta.cpp [hunk 5]
```diff
@@ -22,8 +22,8 @@
 namespace doris {
 
 bool AlphaRowsetMeta::deserialize_extra_properties() {
-    std::string extra_properties = get_extra_properties();
-    bool parsed = _extra_meta_pb.ParseFromString(extra_properties);
+    std::string properties = extra_properties();
+    bool parsed = _extra_meta_pb.ParseFromString(properties);
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_meta.cpp]: May be better to like this?
```
return _extra_meta_pb.ParseFromString(properties); 
```

File: be/src/olap/rowset/alpha_rowset.cpp [hunk 6]
```diff
@@ -105,14 +101,34 @@ bool AlphaRowset::remove_old_files(std::vector<std::string>* removed_links) {
     return true;
 }
 
+int AlphaRowset::get_data_disk_size() const {
+    return _rowset_meta->total_disk_size();
+}
+
+int AlphaRowset::get_index_disk_size() const {
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset.cpp]: May be better to make function name unified?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/alpha_rowset.cpp (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/rowset/alpha_rowset.cpp]: unique_ptr can be used as returned value

**Review by chaoyli — APPROVED** [review]

**Pull request merged**